python/GafferUI/CodeWidget.py

```diff
@@ -555,8 +555,8 @@ def __globalCompletions( self, text ) :
 
 	def __attrAndItemCompletions( self, text ) :
 
-		word = r"[a-zA-Z0-9_]+"
-		optionalWord = r"[a-zA-Z0-9_]*"
+		word = r"[a-zA-Z0-9_:]+"
+		optionalWord = r"[a-zA-Z0-9_:]*"
```

Comment:
`word` gets used in matching both `foo.attr` completions and `foo["item"]` completions, but I think `:` is only valid for the latter?

Reply:
Good point, it may not matter in practice but I've pushed an update that splits out separate `item` and `optionalItem` regexps for the `foo["item"]` completions. We could go further and support matching any valid string for dictionary key completions, but I'm not aware of anyone asking for that...